src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -114,6 +114,13 @@ internal JsonConverter DetermineConverter(Type? parentClassType, Type runtimePro
         /// <returns>
         /// The converter for the given type.
         /// </returns>
+        /// <exception cref="InvalidOperationException">
+        /// The configured <see cref="JsonConverter"/> for <paramref name="typeToConvert"/> returned an invalid converter.
+        /// </exception>
+        /// <exception cref="NotSupportedException">
```

Comment:
Where do the checks happen that result in NSE here?

Reply:
If a converter is not yet cached, it is created from this method and currently the collection-based converters will throw NSE.

Reply:
> he collection-based converters will throw NSE for not supported collections like multidimensional array.

What about the non-collection-based converters/other converters? Is it ever possible that a converter couldn't be created for a particular type (one that we don't support today)? 

For example, what is the exception that we throw if someone happen to have a `BigInteger` property in their model? Presumably it would be this documented NSE. Is that what happens?